src/codegen/codegen.cc

```diff
@@ -211,5 +220,30 @@ std::string PackImportsToC(const runtime::Module& mod, bool system_lib) {
      << "#endif\n";
   return os.str();
 }
+
+runtime::Module PackImportsToLLVM(const runtime::Module& mod,
+                                  bool system_lib,
+                                  const std::string& target) {
+  std::string bin = SerializeModule(mod);
+
+  std::ostringstream os;
+  uint64_t nbytes = bin.length();
+  os << std::hex;
+  for (size_t i = 0; i < sizeof(nbytes); ++i) {
+    os << std::setfill('0') << std::setw(2) << ((nbytes >> (i * 8)) & 0xffUL);
+  }
+  for (size_t i = 0; i < bin.length(); ++i) {
+    int c = bin[i];
+    os << std::setfill('0') << std::setw(2) << (c & 0xff);
+  }
+
+  // Call codegen_blob to generate LLVM module
+  std::string codegen_f_name = "codegen.codegen_blob";
+  // the codegen function.
+  const PackedFunc* codegen_f = runtime::Registry::Get(codegen_f_name);
+  CHECK(codegen_f != nullptr)  << "codegen.codegen_blob is not presented.";
+  return (*codegen_f)(os.str(), system_lib, target);
```

Comment:
Do we need to pass hex here? can we directly pass the binary blob to LLVM?

Reply:
I think we need this. The most important reason is we need this information:
```
  for (size_t i = 0; i < sizeof(nbytes); ++i) {
    os << std::setfill('0') << std::setw(2) << ((nbytes >> (i * 8)) & 0xffUL);
  }
```
Because when we load it back, we will do like this:
```
  uint64_t nbytes = 0;
  for (size_t i = 0; i < sizeof(nbytes); ++i) {
    uint64_t c = mblob[i];
    nbytes |=  (c & 0xffUL) << (i * 8);
  }
```

Reply:
This is only about the way to serialize the bytes in a fixed endian. We should be able to directly append it to binary string and put it in the beginning.

Reply:
std::string header;
for (size_t i = 0; i < sizeof(nbytes); ++i) {
    header.push_back(((nbytes >> (i * 8)) & 0xffUL));
}
then pass header + content

Reply:
After printing the length, inside the PackImportToLLVM, blob is 1044, but inside codegen.codegen_blob, the length of blob is 2.

Reply:
Sorry I forget to mention you cannot pass std string because that will be treated as null terminate c str. Pass a ByteArray instead

Reply:
Cool!